### PR TITLE
add setup node to finalize-release.yaml

### PR DESCRIPTION
**Pull request by DanRod1999**
Add setup node step to pipeline. Otherwise we run the risk of a vm not using a supported version of node.

This caused an error in another pipeline, so we should use node setup or nodeTool whenever possible to avoid this problem

### CODE REVIEW

**Comment by aruniverse** [comment]
this is the only place where we need to do this?
this is a gh pipeline, i was expecting to see changes for ado pipelines

**Comment by DanRod1999** [comment]
> this is the only place where we need to do this? this is a gh pipeline, i was expecting to see changes for ado pipelines

most of our ado pipelines already have this for core already. Otherwise I'm pretty sure we would have experienced breakages from Gytis' PR.
I was just aware that this pipeline might not have it since its GH, so i double checked it.

File: .github/workflows/finalize-release.yaml [hunk 1]
```diff
@@ -27,6 +27,12 @@ jobs:
         run: |
           git config --local user.email 38288322+imodeljs-admin@users.noreply.github.com
           git config --local user.name imodeljs-admin
+
+      - name: Setup node
+        uses: actions/setup-node@v2
```
Comment (aruniverse) [inline .github/workflows/finalize-release.yaml:32]: please update to the use the latest version of these gh actions, 
https://github.com/actions/setup-node/releases

**Comment by aruniverse** [comment]
linking https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md as an fyi re what ver of node is being used by the image

File: .github/workflows/finalize-release.yaml (hunk 1 above)
Comment (aruniverse) [reply .github/workflows/finalize-release.yaml]: please update the other actions and any other gh workflow, 
eg actions/checkout@v2

**Comment by DanRod1999** [comment]
@mergifyio backport release/4.10.x release/4.11.x

**Review by aruniverse — APPROVED** [review]

**Pull request merged**